okio/src/main/kotlin/okio/SegmentPool.kt

```diff
@@ -34,7 +34,7 @@ internal object SegmentPool {
 
   @JvmStatic
   fun take(): Segment {
-    synchronized(SegmentPool::class.java) {
+    synchronized(this) {
```

Comment:
The bytecode here uses the singleton instance, right? I think that's fine. Alternative would be ummm, `SingletonPool::class`, but I think that also does some extra work.